packages/x-telemetry/src/sender.ts

```diff
@@ -0,0 +1,93 @@
+import getTelemetryContext, { TelemetryContextType } from './get-context';
+import { getMuiXTelemetryEnv } from './config';
+import { TelemetryEvent } from './types';
+
+function shouldSendTelemetry(telemetryContext: TelemetryContextType): boolean {
+  // Disable collection of the telemetry
+  // in production environment
+  if (process.env.NODE_ENV === 'production') {
+    return false;
+  }
+
+  // Priority to the config (e.g. in code, env)
+  const envIsCollecting = getMuiXTelemetryEnv('IS_COLLECTING');
+  if (typeof envIsCollecting === 'boolean') {
+    return envIsCollecting;
+  }
+
+  // Disable collection of the telemetry in CI builds,
+  // as it not related to development process
+  if (telemetryContext.traits.isCI) {
+    return false;
+  }
+
+  if (typeof telemetryContext.config.isCollecting === 'boolean') {
+    return telemetryContext.config.isCollecting;
+  }
+
+  // Disabled by default
+  return false;
+}
+
+async function fetchWithRetry(url: string, options: RequestInit, retries = 3): Promise<Response> {
+  try {
+    const response = await fetch(url, options);
+    if (response.ok) {
+      return response;
+    }
+
+    throw new Error(`Request failed with status ${response.status}`);
+  } catch (error) {
+    if (retries === 0) {
+      throw error;
+    }
+
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(fetchWithRetry(url, options, retries - 1));
+      }, Math.random() * 3_000);
+    });
+  }
+}
+
+async function sendMuiXTelemetryEvent(event: TelemetryEvent | null) {
+  try {
```

Comment:
To save more bunde size?

```suggestion
  // Disable collection of the telemetry
  // in production environment
  if (process.env.NODE_ENV === 'production') {
    return;
  }
  try {
```

Actually, it's also important for static security code analysis tools, they will flag a fetch request to an unknown domain without an early return.